.github/workflows/publish-to-pypi.yml

```diff
@@ -35,20 +35,17 @@ on:
   #    - main
 
 jobs:
-  publish-pypi:
-    name: Publish to PyPI
+  build:
+    name: Build distribution 📦
     runs-on: ubuntu-latest
-    permissions:
-      # This permission is mandatory for OIDC publishing
-      id-token: write
-    if: github.repository == 'GenericMappingTools/pygmt'
 
     steps:
     - name: Checkout
       uses: actions/checkout@v4.2.2
       with:
         # fetch all history so that setuptools-scm works
         fetch-depth: 0
+        persist-credentials: false
```

Comment:
Any specific reason to add the `persist-credentials: false` line?

Reply:
This was from the template at https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#checking-out-the-project-and-building-distributions, so I just added it in. The default is `persist-credentials: true` according to https://github.com/actions/checkout/tree/v4.2.2?tab=readme-ov-file#checkout-v4, which would mean the credentials do not persist between jobs (e.g. from the `build` job to the `publish-to-testpypi` and `publish-pypi` jobs and is supposed to be less secure if I'm reading https://github.com/actions/checkout/issues/485 correctly. Setting to `persist-credentials: false` should be more secure, though unsure if it really matters.